### DISCUSSION TITLE
[@types/glob] TS issue with the last version of @types/minimatch

### DISCUSSION THREAD

**Discussion by Malkom** [DefinitelyTyped/DefinitelyTyped#62068] [Issues with a @types package] [answered]
Hi everyone,

I'm not very comfortable with package development but we have a problem with `@types/glob` and `@types/minimatch` on a project.
Our project is in Typescript 2.8.4 and `@types/minimatch` had an update about a week ago that is giving us errors.

In fact, `@types/glob` require `@types/minimatch` with a star: 

```
  "dependencies": {
    "@types/minimatch": "*",
    "@types/node": "*"
  }

```

Would it be possible to lock the version of @types/minimatch to ^3.0.0 to avoid installing v5.1.1?

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment, marked as answer]
you're probably getting errors from unsupported TS features used in e.g. @types/mimimatch.
If you look into NPM details (versions) for each package, you should get TS version comparison, e.g. 2.8 should get 3.0.3 @types/minimatch@3.0.3
https://www.npmjs.com/package/@types/minimatch/v/3.0.3

in NPM you can use overrides:
https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides
in Yarn, resolutions:
https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-how-to-use-it

just lock the versions of those types in your package.json, should fix issue

**Reply by Malkom** [reply]
You are right and finally we found a solution yesterday by upgrading our TS version from 2.8.4 to 3.4.5 !! Just have to make some others tests today so if we have a problem I will saw your answer to overide the version of minimatch.
Thanks !

**Reply by Malkom** [reply]
@peterblazejewicz 

Try to overrides the packages like this : 

```
"overrides": {
    "@types/minimatch": "3.0.5"
  }
```

or 

```
"overrides": {
    "cordova": {
      "cordova-lib": {
        "globby": {
          "@types/glob": {
            "@types/minimatch": "3.0.5"
          }
        }
      }
    }
  }
```
but still install the last version (5.1.2 atm).... Any idea ?
